### PR TITLE
Add import.meta.resolve documentation

### PR DESCRIPTION
**Pull request by mstoykov**
## What?

Add documentation for https://github.com/grafana/k6/pull/3873 and fix eslint to support it.
## Checklist

- [ ] I have used a meaningful title for the PR.
- [ ] I have described the changes I've made in the "What?" section above.
- [ ] I have performed a self-review of my changes.
- [ ] I have run the `npm start` command locally and verified that the changes look good.

- [ ] I have made my changes in the `docs/sources/next` folder of the documentation.
- [ ] I have reflected my changes in the `docs/sources/v{most_recent_release}` folder of the documentation.
- [ ] I have reflected my changes in the relevant folders of the two previous k6 versions of the documentation (if still applicable to previous versions).

- [ ] I have made my changes in the `docs/sources/next` folder of the documentation.

### CODE REVIEW

File: docs/sources/next/javascript-api/_index.md [hunk 1]
```diff
@@ -11,6 +11,14 @@ The list of k6 modules natively supported in your k6 scripts.
 
 {{< docs/shared source="k6" lookup="javascript-api/init-context.md" version="<K6_VERSION>" >}}
 
+## Import.meta
```
Comment (joanlopez) [inline docs/sources/next/javascript-api/_index.md:14]:
```suggestion
## import.meta
```

Should this be lower-cased? 🤔 
And... can we import the table this from `javascript-api/import.meta/_index.md so we dedup?

**Review by joanlopez — APPROVED** [review]
Not sure about ESLint configuration changes, but the rest looks good! 👍🏻 
I left a couple of minor comments, but none is really blocking!

**Comment by mstoykov** [comment]
I don't really like the current way of the dedupping that means that they are shared documents for all the versions and opted to not try to fix this given the time constraint.

cc @heitortsergent - if you can point me how to reuse any file that will be nice.

**Review by codebien — APPROVED** [review]

File: docs/sources/next/javascript-api/_index.md [hunk 2]
```diff
@@ -11,6 +11,14 @@ The list of k6 modules natively supported in your k6 scripts.
 
 {{< docs/shared source="k6" lookup="javascript-api/init-context.md" version="<K6_VERSION>" >}}
 
+## import.meta
+
+`import.meta` is only available in ECMAScript modules, but not CommonJS ones.
+
+| Function                                                                                           | Description                                               |
+| -------------------------------------------------------------------------------------------------- | --------------------------------------------------------- |
+| [import.meta.resolve](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/import.meta/resolve) | resolve path to URL the same way that an ESM import would |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/_index.md:20]:
```suggestion
| [import.meta.resolve](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/import.meta/resolve) | Resolve the path to URL in the same way that an ESM import does. |
```

File: docs/sources/next/javascript-api/import.meta/_index.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+title: 'import.meta'
+description: 'the ECMAScript import meta property'
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/import.meta/_index.md:3]:
```suggestion
description: 'Details about the ECMAScript import meta property.'
```

File: docs/sources/next/javascript-api/import.meta/resolve.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+---
+title: 'resolve( path )'
+description: 'Resolve a path to a URL strign the same way an import would.'
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/import.meta/resolve.md:3]:
```suggestion
description: 'Resolve the path to a URL string in the same way an import statement does.'
```

File: docs/sources/next/javascript-api/import.meta/resolve.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+---
+
+# resolve(path)
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/import.meta/resolve.md:6]:
```suggestion
# resolve( path )
```

We try to keep the title and the h1 heading the same.

File: docs/sources/next/javascript-api/import.meta/resolve.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+Resolve a path to a URL string the same way an import would.
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/import.meta/resolve.md:8]:
```suggestion
Resolve a path to a URL string in the same way an import statement does.'
```

File: docs/sources/next/javascript-api/import.meta/resolve.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+This is useful if you want to provide a path relative to the current module to a function that will use `open` or some other function taking URLs.
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/import.meta/resolve.md:10]:
```suggestion
This is useful if you want to provide a path relative to the current module to a function that uses `open` or another function that takes URLs.
```

File: docs/sources/next/javascript-api/import.meta/resolve.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+This is particularly useful as some functions within k6 are not relative to the same origin, which might lead to seemingly similar paths having very different file being loaded from the filesystem.
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/import.meta/resolve.md:12]:
```suggestion
This is particularly useful as some functions within k6 are not relative to the same origin, which might lead to seemingly similar paths having different files being loaded from the filesystem.
```

File: docs/sources/next/javascript-api/import.meta/resolve.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+For ongoing discussions please see this [issue](https://github.com/grafana/k6/issues/3857).
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/import.meta/resolve.md:14]:
```suggestion
For more details, refer to this [issue](https://github.com/grafana/k6/issues/3857).
```

File: docs/sources/next/javascript-api/import.meta/resolve.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+| Parameter | Type   | Description                |
+| --------- | ------ | -------------------------- |
+| path      | string | a path to a potential file |
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/import.meta/resolve.md:18]:
```suggestion
| path      | string | A file path. |
```

File: docs/sources/next/javascript-api/import.meta/resolve.md [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+### Example
+
+{{< code >}}
+
+```javascript
+const pdfFile = open(import.meta.resolve('./path/to/file.pdf'), 'b');
+
+export default function () {
+  // this is here to not have an exception if ran in k6
```
Comment (heitortsergent) [inline docs/sources/next/javascript-api/import.meta/resolve.md:28]:
```suggestion
  // this is here to avoid an exception if run using k6
```

**Review by heitortsergent — COMMENTED** [review]
Left some small edits, thanks for adding these changes in @mstoykov! 🙇

**Comment by heitortsergent** [comment]
Also @mstoykov I didn't understand your comment about the shared documents for all the versions, could you please explain that to me? 🙏 

We have this tutorial in the Writers' Toolkit about how to create shared content if that helps: https://grafana.com/docs/writers-toolkit/write/reuse-content/reuse-shared-content/

**Comment by mstoykov** [comment]
It seems I got confused by the way it is written and taught the shared files were under `docs/shared` instead of under `docs/sources/<version>/shared` which in that case means that the ones for v50 aren't the same as the one for v053.

So it is all good ! Sorry for the confusion.

**Comment by heitortsergent** [comment]
I think the shared files can be in the `docs/sources/<version>/shared` for version-specific files, or under `docs/sources/shared`. Is that correct @jdbaldry? 🤓

**Comment by jdbaldry** [comment]
> I think the shared files can be in the `docs/sources/<version>/shared` for version-specific files, or under `docs/sources/shared`. Is that correct @jdbaldry? 🤓

I think in theory you can look up in either location. All the directories will be synced to the website repository and you can look up "unversioned" documentation by setting `version=""`.

**Pull request merged**